spec/environment_spec.rb

```diff
@@ -551,4 +551,504 @@ def new_content(id)
+    contentA = create_content
+    contentB = create_content
+    contentC = create_content
+    @cp.add_content_to_product(@owner['key'], product['id'], contentA['id'])
+    @cp.add_content_to_product(@owner['key'], product['id'], contentB['id'])
+    @cp.add_content_to_product(@owner['key'], product['id'], contentC['id'])
+    job = @org_admin.promote_content(@env['id'],[{:contentId => contentB['id'],}])
+    wait_for_job(job['id'], 15)
+    job = @org_admin.promote_content(envB['id'],[{:contentId => contentB['id'],}])
+    wait_for_job(job['id'], 15)
+    job = @org_admin.promote_content(envC['id'],[{:contentId => contentB['id'],}])
+    wait_for_job(job['id'], 15)
+    pool = @cp.create_pool(@owner['key'], product['id'])
+    ent = consumer_cp.consume_pool(pool['id'], {:quantity => 1})[0]
+    x509 = OpenSSL::X509::Certificate.new(ent['certificates'][0]['cert'])
+    extensions_hash = Hash[x509.extensions.collect { |ext| [ext.oid, ext.value] }]
+    serial = ent['certificates'][0]['serial']['serial']
+
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentB['id']}.1")).to be true
+
+    consumer_cp.update_consumer({:environments => [{:name => @env.name}, {:name => envC.name},
+      {:name => envB.name}]})
+    ent = consumer_cp.list_entitlements()[0]
+    x509 = OpenSSL::X509::Certificate.new(ent['certificates'][0]['cert'])
+    extensions_hash = Hash[x509.extensions.collect { |ext| [ext.oid, ext.value] }]
+    new_serial = ent['certificates'][0]['serial']['serial']
+
+    expect(new_serial).to eq(serial)
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentB['id']}.1")).to be true
+  end
+
+  it 'should NOT re-gen consumer entitlement cert when environment priority are reordered and entitlement
+    does not provide specific content' do
+    envB = @org_admin.create_environment(@owner['key'], 'envB', "envB", "For test systems only.")
+    consumer = @org_admin.register(random_string('testsystem'), :system, nil, {}, nil, nil, [], [],
+      [{ 'id' => @env['id']}, {'id' => envB.name}])
+
+    expect(consumer['environments']).not_to be_nil
+
+    consumer_cp = Candlepin.new(nil, nil, consumer['idCert']['cert'], consumer['idCert']['key'])
+    product = create_product
+    contentA = create_content
+    contentB = create_content
+    contentC = create_content
+    contentD = create_content # not added to product
+    @cp.add_content_to_product(@owner['key'], product['id'], contentA['id'])
+    @cp.add_content_to_product(@owner['key'], product['id'], contentB['id'])
+    @cp.add_content_to_product(@owner['key'], product['id'], contentC['id'])
+
+    job = @org_admin.promote_content(@env['id'],[{:contentId => contentB['id'],}])
+    wait_for_job(job['id'], 15)
+    job = @org_admin.promote_content(envB['id'],[{:contentId => contentD['id'],}])
+    wait_for_job(job['id'], 15)
+    job = @org_admin.promote_content(envB['id'],[{:contentId => contentB['id'],}])
+    wait_for_job(job['id'], 15)
```

Comment:
Unsure why you added this, but its good because it has uncovered a bug. If consumer has specified environments @env, envB, which both provide contentB, and their order is reversed (envB, @env), then the _entitlement_ should get regenerated (while right not for this test it isn't). This test is basically the same as the `should re-gen consumer entitlement cert when environment priority are reversed and content originates from different environment` one, but that one is running corrently (while this one isn't). Could it be because of the new changes in `isCertRegenerationRequired`? Anyway, it needs investigation

Reply:
Fixed.
It was due to recent changes to the ```isCertRegenerationRequired``` method.